Website/DesktopModules/AuthenticationServices/DNN/Login.ascx.cs

```diff
@@ -264,17 +264,29 @@ private void OnLoginClick(object sender, EventArgs e)
 
                 //DNN-6093
                 //check if we use email address here rather than username
-                if(PortalController.GetPortalSettingAsBoolean("Registration_UseEmailAsUserName", PortalId, false))
+                UserInfo userByEmail = null;
+                var isUserEmailAsUserName = PortalController.GetPortalSettingAsBoolean("Registration_UseEmailAsUserName", PortalId, false);                
+
+                if (isUserEmailAsUserName)
                 {
-                    var testUser = UserController.GetUserByEmail(PortalId, userName); // one additonal call to db to see if an account with that email actually exists
-                    if(testUser != null)
+                    // one additonal call to db to see if an account with that email actually exists
+                    userByEmail = UserController.GetUserByEmail(PortalId, userName);                     
+
+                    if (userByEmail != null)
                     {
-                        userName = testUser.Username; //we need the username of the account in order to authenticate in the next step
+                        //we need the username of the account in order to authenticate in the next step
+                        userName = userByEmail.Username; 
                     }
                 }
 
-				var objUser = UserController.ValidateUser(PortalId, userName, txtPassword.Text, "DNN", string.Empty, PortalSettings.PortalName, IPAddress, ref loginStatus);
-				var authenticated = Null.NullBoolean;
+                UserInfo objUser = null;
+
+                if (!isUserEmailAsUserName || userByEmail != null)
+                {
```

Comment:
I don't understand why you added this logical statement. Please explain

Reply:
When username not email and isUserEmailasUsername(isUserEmailasUsername) also true, was going into this logic

Reply:
I don't follow. In every case we want to validate the user credential. 

With current code here's what we get

|| isUserEmailAsUserName | userByEmail | userName is | ValidateUser called? | | 
|--| -- | -- | -- | -- | -- | 
|1| true | null | original | no  | ❌ |
|2| true | not null | `userByEmail.Username` | yes | ✅ |
|3| false | null | original | yes | ✅ |
|4| false | not null | original |yes | ✅ |

I'm concerned with row `1`. Please correct me if I'm wrong. I read here that when we enable emails as user names but fail to find a user with that email, the code will never call `UserController.ValidateUser`.

This will be correct if the `Registration_UseEmailAsUserName` means that only email is allowed? Users cannot log in with both email and actually username? @dnnsoftware/tag

Reply:
Yes, exactly It is the case that if the Registration_UserEmailAsUsername is true than login only by email not by username, which is #2367 is about that user was getting logged-in by username once and if you check application is turning username into the email when this UseEmailasusername is true

Reply:
It could make sense but those are significant changes and I don't think worth the trouble and risk